### PR TITLE
`up` should prune manifest

### LINKED ISSUE #1185: Pkg no longer removes unneeded packages from the Manifest file
https://discourse.julialang.org/t/travis-failure-when-testing-required-code-on-julia-v1-0/23986/6

### PR DESCRIPTION
**Pull request by 00vareladavid**
Fixes #1185

### CODE REVIEW

**Comment by 00vareladavid** [comment]
bors try

**Comment by 00vareladavid** [comment]
Still needs a test

File: src/API.jl [hunk 1]
```diff
@@ -170,7 +170,7 @@ function up(ctx::Context, pkgs::Vector{PackageSpec};
 end
 
 resolve(ctx::Context=Context()) =
-    up(ctx, level=UPLEVEL_FIXED, mode=PKGMODE_MANIFEST, update_registry=false)
```
Comment (KristofferC) [inline src/API.jl]: Why change this?

Comment (00vareladavid) [reply src/API.jl]: Passing `PKGMODE_MANIFEST` to `up` will cause `Pkg` to fix all dependencies at their current version, not just the direct dependencies.

**Comment by 00vareladavid** [comment]
bors try

**Comment by 00vareladavid** [comment]
Bors failure is a network error, should be good to merge

File: src/API.jl (hunk 1 above)
Comment (fredrikekre) [reply src/API.jl]: But isn't that what resolve should do?

Comment (00vareladavid) [reply src/API.jl]: Hmm, perhaps I misunderstand the intention behind `resolve`. In any case, that is really a separate issue (we should maybe have a test which checks the difference). I will change it change it back for the sake of merging this PR sooner.

**Comment by 00vareladavid** [comment]
bors try

**Pull request merged**